### DISCUSSION TITLE
Are you able to run vite-plugin-angular through Intellij/Webstorm run configurations?

### DISCUSSION THREAD

**Discussion by svdeveloper1** [analogjs/analog#1380] [Q&A] [answered]
After installing and configuring `@analogjs/vite-plugin-angular`, tests work properly if I run them through the `@analogjs/vitest-angular:test` schematic. If I try to run an individual test through Intellij however, it fails with the error:

![image](https://github.com/user-attachments/assets/c5cc2a4d-4ebc-45f9-9cae-afe2c43aebf6)

```
Error: No test suite found in file [redacted]/app.component.spec.ts
```
This fails even with a bare `describe` block with an empty test. The schematic still succeeds however.
```ts
describe('AppComponent', () => {
  it('should render title', () => {
    // do nothing, should just pass
  });
});
```
If I comment out the `angular()` vite plugin, then it works as expected. 

Is this use case supported? Has anyone used it successfully?

**Comment by svdeveloper1** [comment, marked as answer]
After looking at the example repo, I can see that the problem was the working directory I was using was wrong. It defaulted to the workspace root in an nx monorepo, but the working directory needs to be the same as the project directory. Using a run configuration template solved the issue.

Create a `.run/Template Vitest.run.xml` file in your project directory.
```xml
<component name="ProjectRunConfigurationManager">
  <configuration default="true" type="JavaScriptTestRunnerVitest">
    <config value="$PROJECT_DIR$/apps/<YOUR_APP_NAME_HERE>/vite.config.mts" />
    <node-interpreter value="project" />
    <vitest-package value="$PROJECT_DIR$/node_modules/vitest" />
    <working-dir value="$PROJECT_DIR$/apps/<YOUR_APP_NAME_HERE>" />
    <envs />
    <scope-kind value="ALL" />
    <method v="2" />
  </configuration>
</component>
```

**Reply by brandonroberts** [reply]
Thanks for sharing the solution @svdeveloper1
